x-pack/plugins/security_solution/common/api/timeline/model/components.gen.ts

```diff
@@ -0,0 +1,350 @@
+  eqlOptions: z
+    .object({
+      eventCategoryField: z.string().optional(),
+      tiebreakerField: z.string().optional(),
+      timestampField: z.string().optional(),
+    })
+    .optional(),
+  eventType: z.string().optional(),
+  excludedRowRendererIds: z.array(RowRendererId).optional(),
+  favorite: z.array(FavoriteTimelineResult).optional(),
+  filters: z.array(FilterTimelineResult).optional(),
+  kqlMode: z.string().optional(),
+  kqlQuery: SerializedFilterQueryResult.optional(),
+  indexNames: z.array(z.string()).optional(),
+  savedQueryId: z.string().optional(),
+  sort: Sort.optional(),
+  status: z.enum(['active', 'draft', 'immutable']).optional(),
+  title: z.string().optional(),
+  templateTimelineId: z.string().optional(),
+  templateTimelineVersion: z.number().optional(),
+  timelineType: TimelineType.optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+export type BareNote = z.infer<typeof BareNote>;
+export const BareNote = z.object({
+  eventId: z.string().optional(),
+  note: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+export type Note = z.infer<typeof Note>;
+export const Note = BareNote.merge(
+  z.object({
+    noteId: z.string().optional(),
+    version: z.string().optional(),
+  })
+);
+
+export type PinnedEvent = z.infer<typeof PinnedEvent>;
+export const PinnedEvent = z.object({
+  pinnedEventId: z.string().optional(),
+  eventId: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+  version: z.string().optional(),
+});
+
+export type TimelineResponse = z.infer<typeof TimelineResponse>;
+export const TimelineResponse = SavedTimeline.merge(
+  z.object({
+    eventIdToNoteIds: z.array(Note).optional(),
+    notes: z.array(Note).optional(),
+    noteIds: z.array(z.string()).optional(),
+    pinnedEventIds: z.array(z.string()).optional(),
+    pinnedEventsSaveObject: z.array(PinnedEvent).optional(),
+    savedObjectId: z.string(),
+    version: z.string(),
+  })
+);
+
+export type FavoriteTimelineResponse = z.infer<typeof FavoriteTimelineResponse>;
+export const FavoriteTimelineResponse = z.object({
+  savedObjectId: z.string(),
+  version: z.string(),
+  code: z.number().nullable().optional(),
+  message: z.string().nullable().optional(),
+  templateTimelineId: z.string().nullable().optional(),
+  templateTimelineVersion: z.number().nullable().optional(),
+  timelineType: TimelineType.optional(),
+  favorite: z.array(FavoriteTimelineResult).optional(),
+});
+
+export type GlobalNote = z.infer<typeof GlobalNote>;
+export const GlobalNote = z.object({
+  noteId: z.string().optional(),
+  version: z.string().optional(),
+  note: z.string().optional(),
```

Comment:
This and a few of the fields below are nullable

Reply:
@michaelolo24 I'm actually not sure which schema `GlobalNote` refers to 🤔  There's `BareNote` which is basically the same and it's already defined above.

Reply:
Yea, I'll need to look back. Not sure if it was removed with the most recent notes changes. We can clean this up once we start the integration of the schemas into the actual api's